### PR TITLE
feat: add Erc20 contracts

### PR DESCRIPTION
**Pull request by 0xverin**
### Context

resolves p-845
resolves p-846
This PR mainly implements the ERC20 contract on BSC and Ethereum networks.

### Labels

Please apply following PR-related labels when appropriate:
- `C0-breaking`: if your change could break the existing client, e.g. API change, critical logic change 
- `C1-noteworthy`: if your change is non-breaking, but is still worth noticing for the client, e.g. reference code improvement

### Testing Evidences

generate vc:

erc20(trx):
```
{
	context: ["https://www.w3.org/2018/credentials/v1", "https://w3id.org/security/suites/ed25519-2020/v1"],
	id: "0xdab1d8f33511e9ce4163fc39b069a7d1beed20d9cfb22819cacbe9287df7781c",
	types: [VerifiableCredential],
	credential_subject: CredentialSubject {
		id: "did:litentry:substrate:0x0000000000000000000000000000000000000000000000000000000000000000",
		description: "The amount of a particular token you are holding",
		types: "Token Holding Amount",
		assertion_text: "",
		data_source: None,
		assertions: [And {
			items: [Item {
				src: "$token",
				op: Equal,
				dst: "trx"
			}, Item {
				src: "$holding_amount",
				op: GreaterEq,
				dst: "1"
			}, Item {
				src: "$holding_amount",
				op: LessThan,
				dst: "50"
			}]
		}],
		values: [true],
		endpoint: ""
	},
	issuer: Issuer {
		id: "",
		name: "Litentry TEE Worker",
		mrenclave: "11111111111111111111111111111111",
		runtime_version: IssuerRuntimeVersion {
			parachain: 0,
			sidechain: 0
		}
	},
	issuance_date: "2024-06-13T12:22:46.826837967+00:00",
	parachain_block_number: 0,
	sidechain_block_number: 0,
	proof: None,
	credential_schema: CredentialSchema {
		id: "https://raw.githubusercontent.com/litentry/vc-jsonschema/main/dist/schemas/25-token-holding-amount/1-1-0.json",
		types: "JsonSchemaValidator2018"
	}
}
```

native token(bnb):
```
{
	context: ["https://www.w3.org/2018/credentials/v1", "https://w3id.org/security/suites/ed25519-2020/v1"],
	id: "0x8dc62b246d69406a82c1d2500904480299a1afccbca20b017b7ebb50dabe967e",
	types: [VerifiableCredential],
	credential_subject: CredentialSubject {
		id: "did:litentry:substrate:0x0000000000000000000000000000000000000000000000000000000000000000",
		description: "The amount of a particular token you are holding",
		types: "Token Holding Amount",
		assertion_text: "",
		data_source: None,
		assertions: [And {
			items: [Item {
				src: "$token",
				op: Equal,
				dst: "bnb"
			}, Item {
				src: "$holding_amount",
				op: GreaterEq,
				dst: "1"
			}, Item {
				src: "$holding_amount",
				op: LessThan,
				dst: "50"
			}]
		}],
		values: [true],
		endpoint: ""
	},
	issuer: Issuer {
		id: "",
		name: "Litentry TEE Worker",
		mrenclave: "11111111111111111111111111111111",
		runtime_version: IssuerRuntimeVersion {
			parachain: 0,
			sidechain: 0
		}
	},
	issuance_date: "2024-06-13T12:22:46.834042701+00:00",
	parachain_block_number: 0,
	sidechain_block_number: 0,
	proof: None,
	credential_schema: CredentialSchema {
		id: "https://raw.githubusercontent.com/litentry/vc-jsonschema/main/dist/schemas/25-token-holding-amount/1-1-0.json",
		types: "JsonSchemaValidator2018"
	}
}
```

~~TODO: Add all tokens.~~
All tokens have been added, reference from [rust code](https://github.com/litentry/litentry-parachain/blob/dev/tee-worker/litentry/core/common/src/web3_token/mod.rs#L88).

### CODE REVIEW

**Comment by Kailai-Wang** [comment]
I browsed the whole structure,  it seems to be well desigend and made, nicely done

We'll need @kziemianek and @higherordertech to check more details 🙌

File: tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/erc20/Wbtc.sol [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.8;
+
+import { ERC20 } from "../ERC20.sol";
+import "../../libraries/Identities.sol";
+
+contract SpaceId is ERC20 {
```
Comment (kziemianek) [inline tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/erc20/Wbtc.sol]: Wbtc?

File: tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/NoderealClient.sol [hunk 2]
```diff
@@ -20,58 +20,49 @@ pragma solidity ^0.8.8;
...
-            request,
-            headers
-        );
-        if (result) {
-            return Utils.hexToNumber(balance);
-        } else {
-            return (false, 0);
-        }
-    }
+	function getTokenBalance(
```
Comment (kziemianek) [inline tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/NoderealClient.sol:24]: I wanted it to have 1 function per 1 rpc method but that's fine too.

**Review by kziemianek — APPROVED** [review]

**Comment by kziemianek** [comment]
Can this be further simplified with #2812 ?
If yes then please wait for #2812 to be merged first.

File: tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/NoderealClient.sol (hunk 2 above)
Comment (0xverin) [reply tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/NoderealClient.sol:24]: It is not very convenient to use externally, and requires a lot of if statements, so I merged them here.

**Pull request merged**